ui/components/ui/button/button.component.js

```diff
@@ -53,7 +53,7 @@ const Button = ({
   } else if (submit) {
     buttonProps.type = 'submit';
   }
-  if (typeof buttonProps.onClick === 'function') {
+  if (type === 'link' && typeof buttonProps.onClick === 'function') {
     buttonProps.onKeyUp ??= (event) => {
```

Comment:
Hey @brad-decker : can you plz check if above looks good. It seem this function was added here: https://github.com/MetaMask/metamask-extension/pull/10310/files#diff-011b6626f40e94e6785704e6a069952e76a9fafe727fd293a8fe27b251890395R48

This resulted in a subtle bug where when Enter is pressed on focused buttons `onClick` gets executed twice.

Reply:
Yeah cause the `<button>` has a native enter handler so in effect ours is the downstream one that gets called on the second button rendered. Good catch @jpuri  :D